torch/_inductor/codegen/cpp.py

```diff
@@ -2251,6 +2266,12 @@ def codegen_nodes(self, nodes):
         self.legalize_bf16(nodes)
         self.data_type_propagation(nodes)
 
+        vec_dtype = (
+            torch.bfloat16
+            if all(hasattr(_node, "is_bf16") and _node.is_bf16 for _node in nodes)
```

Comment:
can we check the `dtype` from dtype propagation here instead?

Reply:
the `node` here is `SchedulerNode` but not `fx.Node`

Reply:
I meant whether we can just get the flag by looking into the `fx.Node` in the `SchedulerNode`? Dynamically setting and checking such a flag looks sketchy to me.

Reply:
We have attached the bf16-related information to the `meta` context of `fx.Node`. But it does not cover all the `fx.Node` such as `placeholder`, `get_index`, `index_expr`, and so on. Regarding these operations, its data type might not impact the bf16 scheduler node checker. Regarding the attribute at the runtime, I get your point. Let's consolidate it to the `Optimization` context design.

Reply:
Let me check if we can add a dedicated utility function to make it more elegant.